src/RemoteAPI/class-remote-api-base.php

```diff
@@ -198,4 +198,15 @@ public function is_user_allowed_to_make_api_call(): bool {
 
 		return false;
 	}
+
+	/**
+	 * Returns the request's options for the remote API call.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return array<string, mixed> The array of options.
+	 */
+	protected function get_request_options(): array {
```

Comment:
Kindly add tests to make sure that this is getting called while requesting data from API.

Reply:
Unsure about how we can test this.

Reply:
Add a test spy on it and make sure that it is getting called and correct values are returning.